commit aba8b8ae3cd60839cb863fb750e7944c145f28dc
Author: Mark Robert Miller <markrmiller@apache.org>
Date:   Mon Mar 3 04:35:45 2014 +0000

    SOLR-5761: HttpSolrServer has a few fields that can be set via setters but are not volatile.
    
    git-svn-id: https://svn.apache.org/repos/asf/lucene/dev/trunk@1573442 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/solr/CHANGES.txt b/solr/CHANGES.txt
index e8ca1ed..b2a604c 100644
--- a/solr/CHANGES.txt
+++ b/solr/CHANGES.txt
@@ -110,6 +110,9 @@ Bug Fixes
   streamDeletes is true that should be tucked into the if statement below it.
   (Mark Miller, Gregory Chanan)
 
+* SOLR-5761: HttpSolrServer has a few fields that can be set via setters but
+  are not volatile. (Mark Miller, Gregory Chanan)
+
 Optimizations
 ----------------------
 * SOLR-1880: Distributed Search skips GET_FIELDS stage if EXECUTE_QUERY
diff --git a/solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrServer.java b/solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrServer.java
index eb70701..f1dc3d1 100644
--- a/solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrServer.java
+++ b/solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrServer.java
@@ -86,7 +86,7 @@ public class HttpSolrServer extends SolrServer {
   /**
    * The URL of the Solr server.
    */
-  protected String baseUrl;
+  protected volatile String baseUrl;
   
   /**
    * Default value: null / empty.
@@ -115,14 +115,14 @@ public class HttpSolrServer extends SolrServer {
   
   private final HttpClient httpClient;
   
-  private boolean followRedirects = false;
+  private volatile boolean followRedirects = false;
   
-  private int maxRetries = 0;
+  private volatile int maxRetries = 0;
   
-  private boolean useMultiPartPost;
+  private volatile boolean useMultiPartPost;
   private final boolean internalClient;
 
-  private Set<String> queryParams = Collections.emptySet();
+  private volatile Set<String> queryParams = Collections.emptySet();
 
   /**
    * @param baseURL
